3-building-blocks-of-oop-part-2/hometask/src/Pizza.ts

```diff
@@ -0,0 +1,29 @@
+import { Consumable } from "./Consumable";
+
+export class Pizza extends Consumable {
+    public numberOfSlices: number;
+    private numberOfEatenSlices: number = 0;
+
+    constructor(value: number, weight: number, numberOfSlices: number, isSpoiled: boolean = false) {
+        super('pizza',value, weight, isSpoiled);
+        this.numberOfSlices = numberOfSlices;
+    }
+
+    use(): string {
```

Comment:
that's strange. I see we have base class Consumable which accepts name and has good enough `use` method which can be reused here with `super.use`. I guess it'll log right product name and also, if you set `consumed` to `true` will show right message. However you've fully re-implemented it.

Reply:
Added using super class for `else` block. For `if` block we using own functionality, because messages in super class and pizza class are different.